### PR TITLE
- MessagePack serialization was added

### PR DESCRIPTION
**Pull request by KonstantinRyazantsev**
- Json serialization was fixed and improved

Release notes - https://github.com/LykkeCity/Lykke.RabbitMqDotNetBroker/releases/tag/untagged-a129cc9cda7584495a29

### CODE REVIEW

File: src/Lykke.RabbitMqBroker/Subscriber/FormatMigrationMessageDeserializerFactory.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    /// Factory of the typical format migration message deserializers
+    /// </summary>
+    [PublicAPI]
+    public static class FormatMigrationMessageDeserializerFactory
+    {
+        /// <summary>
+        /// Creates <see cref="FormatMigrationMessageDeserializer{TMessage}"/> for migration from the Json to the MessagePack
+        /// </summary>
+        public static FormatMigrationMessageDeserializer<TMessage> JsonToMessagePack<TMessage>(
+            Encoding jsonEncoding = null,
```
Comment (starkmsu) [inline src/Lykke.RabbitMqBroker/Subscriber/FormatMigrationMessageDeserializerFactory.cs]: Why do we need all these c-tor params that are not used in c-tor body?

File: src/Lykke.RabbitMqBroker/Subscriber/FormatMigrationMessageDeserializer.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            _primaryDeserializer = _newFromatDeserializer;
+        }
+
+        public TMessage Deserialize(byte[] data)
+        {
+            try
+            {
+                return _primaryDeserializer.Deserialize(data);
+            }
+            catch
```
Comment (starkmsu) [inline src/Lykke.RabbitMqBroker/Subscriber/FormatMigrationMessageDeserializer.cs]: I think that in some cases thee won't be an exception here on problems with deserialization - default(TMessage) might be returned as well. Such check should be added.

Comment (KonstantinRyazantsev) [reply src/Lykke.RabbitMqBroker/Subscriber/FormatMigrationMessageDeserializer.cs]: If deserializer can't deserialize the message, it should throws an exception. If we will check for default(TMessage) here, we can't unequivocally say was that error or actual deserialized value. Especially for the structs.

File: src/Lykke.RabbitMqBroker/Subscriber/MessagePackMessageDeserializer.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        {
+            _formatterResolver = formatterResolver;
+            _readStrict = readStrict;
+        }
+
+        public TMessage Deserialize(byte[] data)
+        {
+            using (var stream = new MemoryStream(data))
+            {
+                stream.Position = 0;
```
Comment (MikhailGerasimov) [inline src/Lykke.RabbitMqBroker/Subscriber/MessagePackMessageDeserializer.cs]: You don't have to set the position to 0

**Pull request merged**